release_dashboard/lib/logic/git_regex.dart

```diff
@@ -0,0 +1,20 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+const String gitHashPattern = r'[0-9a-f]{40}';
+
+// Allows empty string because the dart revision which is the only input using this regex could be empty
+final RegExp gitHashRegex = RegExp([r'^$|^', gitHashPattern, r'$'].join());
+
+/// Supports multiple git hashes delimited by a comma or an empty string.
+///
+/// valid: c7714158950347,c7714158950347,c7714158950347
+/// valid: c7714158950347
+/// unvalid:   c7714158950347,@@cccccc
+/// unvalid (cannot end with a comma):   c7714158950347,c7714158950347,
+var multiGitHashPattern = [r'^$|^', gitHashPattern, r'(?:,', gitHashPattern, r')*$'].join();
+
+final RegExp multiGitHashRegex = RegExp(multiGitHashPattern);
+
+final RegExp candidateBranchRegex = RegExp(r'^flutter-(\d+)\.(\d+)-candidate\.(\d+)$');
```

Comment:
make sure you're calling `.trim()` on all inputs you compare against these regexes, since you're using `^` and `$`

Reply:
Why not?

Reply:
+1 - trim() is pretty quick to call, and improves the UX. It's common to do input sanitization for common user issues like this